File: C/longest-common-prefix.c
#include <stdio.h>
#include <stdlib.h>

char* longestCommonPrefix(char** strs, int strsSize) {
    char* prefix=(char*)calloc(1,200);
    if(strsSize==1){
        return strs[0];
    }
    for(int i=0;strs[i]!='\0'&&i<200;i++){
        char c = strs[0][i];
        for(int j=1;i<strsSize;j++){
            if('\0'==strs[j][i]||c!=strs[j][i]){
                return prefix;
            }
        }
        prefix[i]=c;
    }
    return prefix;
}
int main() {
    char* strs[] = {"flower", "flow", "flight"};
    int strsSize = 3;
    char* result = longestCommonPrefix(strs, strsSize);
    printf("Longest Common Prefix: %s\n", result);
    free(result);
    return 0;
}

File: Python/longest-common-prefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        i = 0
        len_strs = len(strs)
        lens_strs = [len(s) for s in strs]

        def equal_at_i(i: int) -> bool:
            if lens_strs[0] <= i:
                return False
            strs0_at_i = strs[0][i]
            for j in range(1, len_strs):
                if i >= lens_strs[j] or strs[j][i] != strs0_at_i:
                    return False
            return True

        while equal_at_i(i):
            i += 1
        return strs[0][:i]


if __name__ == "__main__":
    strs = ["flower", "flow", "flight"]
    sol = Solution()
    print(sol.longestCommonPrefix(strs))
